parser-typechecker/src/Unison/Runtime/Machine.hs

```diff
@@ -2159,7 +2159,7 @@ universalCompare frn = cmpc False
     cmpc tyEq (Foreign fl) (Foreign fr)
       | Just sl <- maybeUnwrapForeign Rf.listRef fl,
         Just sr <- maybeUnwrapForeign Rf.listRef fr =
-          comparing Sq.length sl sr <> fold (Sq.zipWith (cmpc tyEq) sl sr)
+          fold (Sq.zipWith (cmpc tyEq) sl sr)
```

Comment:
I think this isn't correct - it will have `[1,2,3]` equal to `[1,2,3,4]`, right? Since the zipWith cuts off at the shorter element.

I think putting the length comparison after the `zipWith` comparison would do it, but could you add a few tests to `builtins.md` transcript?

Reply:
Oh right. Good catch.